concent_api/core/tests/test_core_views.py

```diff
@@ -929,3 +933,108 @@ def test_view_receive_out_of_band_return_http_204_if_no_messages_in_database(sel
+                url='core:send',
+                data=dump(
+                    self.correct_golem_data,
+                    self.PROVIDER_PRIVATE_KEY,
+                    CONCENT_PUBLIC_KEY),
+                golem_messages_version='1.12.0'
+            )
+        self._test_response(
+            response,
+            status=200,
+            key=self.PROVIDER_PRIVATE_KEY,
+            message_type=message.concents.ServiceRefused,
+            fields={
+                'reason': message.concents.ServiceRefused.REASON.InvalidRequest,
+            }
+        )
+        self.assertIn('Wrong version of golem messages. Clients version is 1.12.0, Concent version is 2.15.0', str(log_mock.call_args))
+
+    def test_that_concent_should_accept_request_without_protocol_version(self):
+        response = self.client.post(
+            reverse('core:send'),
+            data=dump(
+                self.correct_golem_data,
+                self.PROVIDER_PRIVATE_KEY,
+                CONCENT_PUBLIC_KEY),
+            content_type='application/octet-stream',
+        )
+        self._test_response(
+            response,
+            status=202,
+            key=self.PROVIDER_PRIVATE_KEY,
+
+        )
+
+    def test_that_send_should_refuse_request_if_all_stored_messages_have_incompatible_protocol_version(self):
+        with override_settings(GOLEM_MESSAGES_VERSION='1.11.0'):
+            store_subtask(
+                task_id=self.task_to_compute.compute_task_def['task_id'],
+                subtask_id=self.task_to_compute.compute_task_def['subtask_id'],
+                provider_public_key=self.provider_public_key,
+                requestor_public_key=self.requestor_public_key,
+                state=Subtask.SubtaskState.FORCING_REPORT,
+                next_deadline=int(self.task_to_compute.compute_task_def['deadline']) + settings.CONCENT_MESSAGING_TIME,
+                task_to_compute=self.task_to_compute,
+                report_computed_task=self.report_computed_task,
+            )
+        with mock.patch('core.utils.log') as log_not_called_mock, \
+                mock.patch('core.subtask_helpers.log') as log_called_mock:
+            response = self.send_request(
+                url='core:send',
+                data=dump(
+                    self.correct_golem_data,
+                    self.PROVIDER_PRIVATE_KEY,
+                    CONCENT_PUBLIC_KEY),
+            )
+        self._test_response(
+            response,
+            status=200,
+            key=self.PROVIDER_PRIVATE_KEY,
+            message_type=message.concents.ServiceRefused,
+            fields={
+                'reason': message.concents.ServiceRefused.REASON.InvalidRequest,
+            }
+        )
+
+        log_called_mock.assert_called()
+        log_not_called_mock.assert_not_called()
+
+        self.assertIn(f'Version stored in database is 1.11.0, Concent version is {settings.GOLEM_MESSAGES_VERSION}', str(log_called_mock.call_args))
```

Comment:
I think that all these new tests should be in new file

Reply:
Why? It concerns core_views so it is in test_core_views